### PR TITLE
Added fileschema and filecontroller

### LINKED ISSUE #16: Add files (maps and tilesets) API

### PR DESCRIPTION
**Pull request by ghost**
Added a file schema with only data fields
Added a file controller with a default route setup but not implemetations

### CODE REVIEW

**Comment by ghost** [comment]
I have added only file schemas. No tests are added since no apis are added. 
Please approve when you need file schemas to work on your parts!

File: resources/file/fileSchema.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    maxLength: [20, 'Username can be at most 20 characters long'],
+    set: v => v.toLowerCase(),
+  },
+  name: {
+    type: String,
+    required: [true, 'File name is required'],
+  },
+  comments: {
+    type: [CommentSchema],
+    default: [],
```
Comment (lincolnnguyen18) [inline resources/file/fileSchema.js:38]: Think arrays are empty by default in Mongoose

File: resources/file/fileSchema.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import mongoose, { Schema } from 'mongoose';
+
+const CommentSchema = new Schema({
+  authorUsername: {
+    type: String,
+    required: [true, 'Authoer\'s username is required'],
```
Comment (lincolnnguyen18) [inline resources/file/fileSchema.js:6]: Typo

**Review by lincolnnguyen18 — CHANGES_REQUESTED** [review]
Looks mostly good! Could you also rebase on the latest commit on main? It is https://github.com/team-orange-cse416/tiletogether-service/commit/ccdb01d3012eaa202a122118dff77b1d686b6549

**Review by shohanchowdhury — APPROVED** [review]

**Pull request closed without merging**